plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/preferences/OpenShiftCorePreferences.java

```diff
@@ -61,7 +63,11 @@ private StringPreferenceValue createPreferenceValue(String connectionURL) {
 	}
 	
 	public String getOCBinaryLocation() {
-		return ocBinaryLocation.get();
+		String result = ocBinaryLocation.get();
+		if(StringUtils.isEmpty(result)) {
```

Comment:
you mean the method to find the oc binary on path, right?
I called it #findLocation(). Works for you?